packages/spacecat-shared-rum-api-client/src/common/aggregateFns.js

```diff
@@ -52,6 +54,59 @@ function getCTRByUrl(bundles) {
     }, {});
 }
 
+/**
+ * Calculates the Click-Through Rate (CTR) by URL and Referrer.
+ * CTR is defined as the total number of sessions with at least one click event per referrer.
+ * divided by the total number of pageviews for each URL per referrer.
+ *
+ * @param {Array<Object>} bundles - An array of RUM bundles (NOT Flat bundles).
+ * @returns {Object} - An object where the key is the URL and the value is an object
+ * with the CTR value by referrer.
+ */
+function getCTRByUrlAndChannel(bundles) {
```

Comment:
everytime when we are aggregating, we are essentially filtering and reducing the same "bundles". I was wondering if we can make the existing `getCTRByUrl` method more genering where we pass the filter and checkpoint to calculate the average in more generic way instead of adding a new method?

Reply:
@ekremney 
> we are essentially filtering and reducing the same "bundles"

Yes, that's a bit of additional computing.

> I was wondering if we can make the existing getCTRByUrl method more genering where we pass the filter and checkpoint to calculate the average in more generic way instead of adding a new method?

We're not calling `getCTRByUrl` now, only calling `getCTRByUrlAndVendor`, so no performance impact from the previous implementation. But a generic way would be good, but for now, I'd like to skip.